### PR TITLE
Update Azure & AWS Provider version

### PR DESCRIPTION
**Pull request by sumanthreddy29**
This pull request includes several changes to the Terraform configurations for both AWS and Azure, as well as updates to a YAML job file. The most important changes include version updates for required providers, modifications to Azure Kubernetes Service (AKS) configurations, and the commenting out of certain job steps in the YAML file.

### Provider Version Updates:
* [`modules/terraform/aws/versions.tf`](diffhunk://#diff-9c29578937d87994f79600b40d6b6a0219023626d9f4b1f4a045091a2d60b278L6-R6): Updated the required AWS provider version from `<= 5.38` to `<= 5.73`.
* [`modules/terraform/azure/versions.tf`](diffhunk://#diff-db80da69c1925b208c7e0de8f495645e88cce4aee7e92412bc60cd5aec7c4a06L6-R6): Updated the required AzureRM provider version from `<= 3.93.0` to `<= 4.6.0`.

### AKS Configuration Changes:
* [`modules/terraform/azure/aks/main.tf`](diffhunk://#diff-4089bc0576048d82e335da51ad0374e2e9c0ae37db3d65c3c784e97871651fd7L37-R37): Replaced `ebpf_data_plane` with `network_data_plane` in the AKS resource configuration.
* [`modules/terraform/azure/aks/variables.tf`](diffhunk://#diff-e0b25de6470cd5774403141e8997adee371642381fd8430d6dd467f86d13b59eL47-R47): Updated the `aks_config` variable to replace `ebpf_data_plane` with `network_data_plane` and added a `revisions` field to the `service_mesh_profile` object. [[1]](diffhunk://#diff-e0b25de6470cd5774403141e8997adee371642381fd8430d6dd467f86d13b59eL47-R47) [[2]](diffhunk://#diff-e0b25de6470cd5774403141e8997adee371642381fd8430d6dd467f86d13b59eR78)
* [`modules/terraform/azure/variables.tf`](diffhunk://#diff-a70d527b3b015f3aa3103f024416bbf9c1e2828c82890a61a97316c23e5402efL66-R72): Updated the `aks_config_list` variable to replace `ebpf_data_plane` with `network_data_plane` and added a `revisions` field to the `service_mesh_profile` object.

### CODE REVIEW

File: .github/workflows/terraform-validation.yml [hunk 1]
```diff
@@ -134,12 +134,9 @@ jobs:
 
       - name: Get job id and set env
         run: |
-          echo "RUN_ID=123456789" >> "$GITHUB_ENV"
```
[the remark this answers is not in the document]
Comment (sumanthreddy29) [reply .github/workflows/terraform-validation.yml:137]: It's not used anymore

File: steps/terraform/run-command.yml [hunk 2]
```diff
@@ -86,8 +86,8 @@ steps:
   env:
     REGIONS: ${{ convertToJson(parameters.regions) }}
     CLOUD: ${{ parameters.cloud }}
+    ARM_SUBSCRIPTION_ID: $(AZURE_SUBSCRIPTION_ID)
```
[the remark this answers is not in the document]
Comment (sumanthreddy29) [reply steps/terraform/run-command.yml:89]: From azure provider version 4 we need to pass azure subscription id for terraform, so i made it default for both managed identity and service connection.

[the remark this answers is not in the document]
Comment (sumanthreddy29) [reply steps/terraform/run-command.yml:89]: https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/guides/4.0-upgrade-guide#specifying-subscription-id-is-now-mandatory

File: modules/terraform/azure/main.tf [hunk 3]
```diff
@@ -25,7 +25,7 @@ locals {
           aks.network_profile,
           {
             network_policy  = local.aks_network_policy != null ? local.aks_network_policy : aks.network_profile.network_policy
-            ebpf_data_plane = local.aks_network_dataplane != null ? local.aks_network_dataplane : aks.network_profile.ebpf_data_plane
+            ebpf_data_plane = local.aks_network_dataplane != null ? local.aks_network_dataplane : aks.network_profile.network_data_plane
```
[the remark this answers is not in the document]
Comment (sumanthreddy29) [reply modules/terraform/azure/main.tf]: Updated

File: modules/terraform/azure/variables.tf [hunk 4]
```diff
@@ -63,12 +63,13 @@ variable "aks_config_list" {
       network_plugin      = optional(string, null)
       network_plugin_mode = optional(string, null)
       network_policy      = optional(string, null)
-      ebpf_data_plane     = optional(string, null)
+      network_data_plane  = optional(string, null)
```
Comment (anson627) [inline modules/terraform/azure/variables.tf:66]: nit: dataplane can be one word

**Review by alyssa1303 — APPROVED** [review]

**Pull request merged**